### PR TITLE
DCON-3407 Add AuditEvent migration script: Atlas Online Archive to ClickHouse

### PR DESCRIPTION
**Pull request by mintusah25**
## Summary

- Adds a standalone Node.js migration script to bulk-transfer ~55TB of AuditEvent data from Atlas Online Archive (via Data Federation) to ClickHouse
- Processes ~1,553 daily partitions with configurable concurrency (default 6 workers), resume-safe checkpointing via a ClickHouse state table, and per-day count verification for zero data loss
- Includes CLI modes: `--dry-run`, `--resume`, `--verify-only` for safe incremental operation

## New Files

| File | Purpose |
|------|---------|
| `clickhouse-init/03-audit-event-migration-state.sql` | MergeTree state table DDL |
| `src/scripts/migrate_audit_events_to_clickhouse.js` | CLI entry point + worker pool |
| `src/scripts/lib/migrationStateManager.js` | ClickHouse state CRUD |
| `src/scripts/lib/auditEventTransformer.js` | MongoDB doc → ClickHouse row |
| `src/scripts/lib/partitionWorker.js` | Single-day cursor → transform → insert |
| `src/scripts/lib/migrationVerifier.js` | Per-day count verification |
| `src/tests/scripts/migrate_audit_events_to_clickhouse.test.js` | 12 unit tests |

## Test plan

- [x] 12 unit tests pass (transformer, partition generation, edge cases)
- [x] Lint passes clean
- [x] Run with `--dry-run` against Atlas Data Federation to verify connectivity and partition seeding
- [x] Run small date range (e.g., 1 week) to validate end-to-end flow
- [x] Run `--verify-only` to confirm per-day count match
- [x] Full migration run with `--concurrency 6`

### CODE REVIEW

File: src/admin/utils/migrationStateManager.js [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+        return newDays.length;
+    }
+
+    /**
+     * Gets all partition states
+     * @returns {Promise<Array<{partition_day: string, status: string, source_count: number, inserted_count: number, last_mongo_id: string, error_message: string}>>}
+     */
+    async getAllStatesAsync() {
+        return this.clickHouseClientManager.queryAsync({
+            query: `SELECT * FROM ${this.table} FINAL ORDER BY partition_day`
```
[the remark this answers is not in the document]
Comment (mintusah25) [reply src/admin/utils/migrationStateManager.js:62]: Just migration script, leave it for now

File: src/admin/utils/migrationStateManager.js [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+    }
+
+    /**
+     * Gets state for a single partition day
+     * @param {string} partitionDay
+     * @returns {Promise<Object|undefined>}
+     */
+    async getStateForDayAsync(partitionDay) {
+        const results = await this.clickHouseClientManager.queryAsync({
+            query: `SELECT * FROM ${this.table} FINAL
```
[the remark this answers is not in the document]
Comment (mintusah25) [reply src/admin/utils/migrationStateManager.js:73]: Same here

File: src/admin/utils/partitionWorker.js [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+            recorded: { $gte: dayStart.toISOString(), $lt: dayEnd.toISOString() }
+        };
+        const sourceCount = await collection.countDocuments(sourceCountQuery);
+
+        if (sourceCount === 0) {
+            return { insertedCount: 0, sourceCount: 0, skippedCount: 0 };
+        }
+
+        await this.stateManager.markInProgressAsync(partitionDay);
+
```
[the remark this answers is not in the document]
Comment (mintusah25) [reply src/admin/utils/partitionWorker.js:85]: Leave it for now

File: src/admin/scripts/migrateAuditEventsToClickhouse.js [hunk 4]
```diff
@@ -0,0 +1,417 @@
...
+        mongoUrl = mongoUrl.replace(
+            'mongodb://',
+            `mongodb://${env.AUDIT_EVENT_MONGO_USERNAME}:${env.AUDIT_EVENT_MONGO_PASSWORD}@`
+        );
+    }
+
+    mongoUrl = encodeURI(mongoUrl);
+    const dbName = env.AUDIT_EVENT_MONGO_DB_NAME || 'fhir';
+
+    return { mongoUrl, dbName };
```
[the remark this answers is not in the document]
Comment (mintusah25) [reply src/admin/scripts/migrateAuditEventsToClickhouse.js:80]: Fixed

**Review by shubhamgoel24 — APPROVED** [review]
